### ISSUE TITLE
`textDocument/semanticTokens/full/delta` doesn't work

### ISSUE THREAD

**Issue by fannheyward** [rust-lang/rust-analyzer#8253]
Hello, from initialize capabilities, rust-analyzer responses with `full: {delta: true}`, but in my tests, rust-analyzer doesn't response for delta request:

Full request:

```
Sending request 'textDocument/semanticTokens/full - (22)'.
Params: {
    "textDocument": {
        "uri": "file:///Users/fannheyward/src/hello-rust/src/main.rs"
    }
}
```

Full response:

```
Received response 'textDocument/semanticTokens/full - (22)' in 6886ms.
Result: {
    "resultId": "1",
    "data": [
        0,
        0,
        2,...
    ]
}
```

Delta request:

```
Sending request 'textDocument/semanticTokens/full/delta - (37)'.
Params: {
    "textDocument": {
        "uri": "file:///Users/fannheyward/src/hello-rust/src/main.rs"
    },
    "previousResultId": "1"
}
```

Delta response, _still full response_:

```
Received response 'textDocument/semanticTokens/full/delta - (37)' in 21ms.
Result: {
    "resultId": "2",
    "data": [
        0,
        0,
        2,...
    ]
}
```

**Comment by kjeremy** [comment]
Can you show your tests? Also are you saying that it's responding with the
incorrect type or that the delta contains the full set of tokens? The spec
allows for the full set to be returned if it cannot compute a delta for
some reason. One reason for this would be that the client has requested a
refresh or we have reloaded the workspace.

**Comment by fannheyward** [comment]
> The spec allows for the full set to be returned if it cannot compute a delta for some reason

I missed this.

**Comment by kjeremy** [comment]
To be clear... Does it eventually report a delta? I do want to make sure
that it's working.

**Comment by fannheyward** [comment]
No. Client requests `textDocument/semanticTokens/full`, RA returns full response, after some code changed, client requests `textDocument/semanticTokens/full/delta`, but RA still returns full response.

**Comment by kjeremy** [comment]
That's concerning. Can you post a test or something that reproduces the
issue?

**Comment by fannheyward** [comment]
1. `cargo new helloworld` and cd into
2. `code .`, enable `"rust-analyzer.trace.server": "verbose"`
3. search `semanticTokens/full` in rust-analyzer language server trace logs
4. the first time, RA returns full tokens
5. make any changes, change `Hello, world` to `Hello`
6. client requests `textDocument/semanticTokens/full/delta`, but RA still returns full tokens

**Comment by AhoyISki** [comment]
Any updates on this? I'm having the same issue.
